src/saml2/cert.py

```diff
@@ -336,31 +340,13 @@ def verify(self, signing_cert_str, cert_str):
             cert_algorithm = cert.get_signature_algorithm()
             if six.PY3:
                 cert_algorithm = cert_algorithm.decode('ascii')
+                cert_str = cert_str.encode('ascii')
```

Comment:
`cert_str` is a PEM formatted string here, which is guaranteed to be ascii. The requirement for this is because `load_pem_x509_certificate` takes bytes, not text.